src/Workspaces/Core/Portable/Classification/ClassificationService.cs

```diff
@@ -1,25 +1,75 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
-using System.Collections.Generic;
 using System.Collections.Immutable;
-using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Classification.Classifiers;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Classification
 {
+    /// <summary>
+    /// A language service that identifies classifications for ranges of a <see cref="Document"/>.
+    /// </summary>
     internal abstract class ClassificationService : ILanguageService
     {
-        // this is really GetQuickClassifications
-        public abstract ImmutableArray<ClassifiedSpan> GetLexicalClassifications(SourceText text, TextSpan span, CancellationToken cancellationToken);
+        /// <summary>
+        /// Gets the appropriate <see cref="ClassificationService"/> for the specified <see cref="Document"/> .
+        /// </summary>
+        public static ClassificationService GetService(Document document)
+        {
+            return GetService(document.Project.Solution.Workspace, document.Project.Language);
+        }
 
-        public abstract ClassifiedSpan AdjustClassification(SourceText text, ClassifiedSpan classifiedSpan);
+        /// <summary>
+        /// Gets the appropriate <see cref="ClassificationService"/> for the specified <see cref="Workspace"/> and language.
+        /// </summary>
+        public static ClassificationService GetService(Workspace workspace, string language)
+        {
+            return workspace.Services.GetLanguageServices(language).GetService<ClassificationService>() ?? NoOpService.Instance;
```

Comment:
We should do this for all services that are consumed publicly. 

---
In reply to: [78663587](https://github.com/dotnet/roslyn/pull/13741#discussion_r78663587) [](ancestors = 78663587)